python/pyspark/sql/connect/types.py

```diff
@@ -41,6 +41,15 @@
     BooleanType,
     NullType,
 )
+import decimal
```

Comment:
While I generally agree with the refactoring, I think it would be good to split this into two PRs. One that does the refactoring and then this one. Otherwise this just combines unrelated changes into one PR.